x/axelarnet/handler.go

```diff
@@ -91,3 +98,47 @@ func NewHandler(k keeper.Keeper, n types.Nexus, b types.BankKeeper, a types.Acco
 		return res, nil
 	}
 }
+
+// NewProposalHandler returns the handler for the proposals of the axelarnet module
+func NewProposalHandler(nexusK types.Nexus, accountK types.AccountKeeper) govtypes.Handler {
+	return func(ctx sdk.Context, content govtypes.Content) error {
+		switch c := content.(type) {
+		case *types.CallContractsProposal:
+			for _, contractCall := range c.ContractCalls {
+				sender := nexus.CrossChainAddress{Chain: exported.Axelarnet, Address: accountK.GetModuleAddress(govtypes.ModuleName).String()}
+				recipient := nexus.CrossChainAddress{Chain: funcs.MustOk(nexusK.GetChain(ctx, contractCall.Chain)), Address: contractCall.ContractAddress}
+				// axelar gateway expects keccak256 hashes for payloads
+				payloadHash := crypto.Keccak256(contractCall.Payload)
+
+				msgID, txID, nonce := nexusK.GenerateMessageID(ctx)
```

Comment:
txID is probably zero bytes since this is run in the end blocker? i guess that's an ok default

Reply:
yes